### PR TITLE
[SEDONA-411] Add RS_Rotation

### PR DESCRIPTION
**Pull request by furqaankhan**
## Did you read the Contributor Guide?

- Yes, I have read [Contributor Rules](https://sedona.apache.org/latest-snapshot/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/latest-snapshot/community/develop/)

## Is this PR related to a JIRA ticket?

- Yes, the URL of the associated JIRA ticket is https://issues.apache.org/jira/browse/SEDONA-411. The PR name follows the format `[SEDONA-XXX] my subject`.

## What changes were proposed in this PR?

- Add RS_Rotation
- Add RS_GeoTransform

## How was this patch tested?

- Passed new tests

## Did this PR include necessary documentation updates?

- Yes, I have updated the documentation update.

### CODE REVIEW

**Comment by jiayuasu** [comment]
@furqaankhan Is the result identical to PostGIS?

**Comment by furqaankhan** [comment]
Yes it is.

**Comment by furqaankhan** [comment]
@jiayuasu I want to get Kristin's opinion before merging, so don't merge yet.

File: common/src/main/java/org/apache/sedona/common/raster/RasterAccessors.java
Comment (jiayuasu) [inline common/src/main/java/org/apache/sedona/common/raster/RasterAccessors.java:173]: can `magnitude` be zero or other invalid values? If so, we need to add handlers and test cases.

**Review by jiayuasu — CHANGES_REQUESTED** [review]

File: common/src/main/java/org/apache/sedona/common/raster/RasterAccessors.java
Comment (Kontinuation) [inline common/src/main/java/org/apache/sedona/common/raster/RasterAccessors.java:151]: This is not correct. `offsetX` and `offsetY` should be directly retrieved from the upperLeftX and upperLeftY components of the raster metadata.

File: common/src/test/java/org/apache/sedona/common/raster/RasterAccessorsTest.java
Comment (Kontinuation) [inline common/src/test/java/org/apache/sedona/common/raster/RasterAccessorsTest.java:74]: We are getting 0 offsets for rasters with non-zero offsets, this is caused by the problem mentioned above.

**Review by Kontinuation — CHANGES_REQUESTED** [review]

**Review by jiayuasu — APPROVED** [review]

**Pull request merged**
